client/app/shared/modals/model-edit/model-edit.component.ts

```diff
@@ -49,4 +49,8 @@ export class ModelEdit {
     return ['text', 'time'].some(type => type === checkedType)
   }
 
+  trackByField (index, field) {
```

Comment:
I would change the method name to indicate by which field we are tracking by. e.g. `trackByDisplay`